gravitee-am-ui/src/app/domain/applications/application/advanced/factors/mfa-challenge/mfa-challenge.component.ts

```diff
@@ -1,15 +1,183 @@
+    this.challenge.active = !this.challenge.active;
+    this.update();
+  }
+
+  selectChallengeOption(option: ModeOption): void {
+    this.currentMode = option;
+    this.update();
+  }
+
+  isChallengeChecked(option: ModeOption): boolean {
+    return this.currentMode === option;
+  }
+
+  updateRiskAssessment(assessments: any): void {
+    this.assessments = assessments;
+    this.update();
+  }
+
+  updateConditional($event: any): void {
+    this.challenge.challengeRule = $event.target.value;
+    this.update();
+  }
+
+  private getAssessment(riskAssessment: any, assessmentName: string): any {
+    return riskAssessment?.[assessmentName] ? riskAssessment[assessmentName] : { enabled: false, thresholds: {} };
+  }
+
+  private getRiskAssessment(riskAssessment: any, enabled: boolean): any {
+    return {
+      enabled: enabled,
+      deviceAssessment: this.getAssessment(riskAssessment, 'deviceAssessment'),
+      ipReputationAssessment: this.getAssessment(riskAssessment, 'ipReputationAssessment'),
+      geoVelocityAssessment: this.getAssessment(riskAssessment, 'geoVelocityAssessment'),
+    };
+  }
+
+  private computeRiskAssessmentRule(riskAssessment: any): string {
+    let rule = '{';
+    const devices = riskAssessment.deviceAssessment;
+    if (devices?.enabled) {
+      rule += MfaChallengeComponent.RISK_ASSESSMENT_PREFIX + 'devices' + MfaChallengeComponent.RISK_ASSESSMENT_SUFFIX;
+    }
+    const ipReputation = riskAssessment.ipReputationAssessment;
+    if (ipReputation?.enabled) {
+      rule +=
+        (rule.length === 1 ? ' ' : ' && ') +
+        MfaChallengeComponent.RISK_ASSESSMENT_PREFIX +
+        'ipReputation' +
+        MfaChallengeComponent.RISK_ASSESSMENT_SUFFIX;
+    }
+    const geoVelocity = riskAssessment.geoVelocityAssessment;
+    if (geoVelocity?.enabled) {
+      rule +=
+        (rule.length === 1 ? ' ' : ' && ') +
+        MfaChallengeComponent.RISK_ASSESSMENT_PREFIX +
+        'geoVelocity' +
+        MfaChallengeComponent.RISK_ASSESSMENT_SUFFIX;
+    }
+    rule += '}';
+    return rule;
+  }
+
+  private initModes(): void {
+    this.modes = Object.keys(MfaChallengeComponent.modeOptions).map((key) => {
+      const option = MfaChallengeComponent.modeOptions[key];
+      option.isMissingFeature$ = this.licenseService.isMissingFeature$(option.licenseOptions);
+      return option;
+    });
+  }
+
+  private update(): void {
+    const isRiskBased = this.currentMode === MfaChallengeComponent.modeOptions.RISK_BASED;
+    this.riskAssessment = this.getRiskAssessment(this.assessments, isRiskBased);
+    if (isRiskBased) {
+      this.adaptiveMfaRule = this.computeRiskAssessmentRule(this.riskAssessment);
+    } else if (this.currentMode === MfaChallengeComponent.modeOptions.CONDITIONAL) {
+      this.adaptiveMfaRule = this.challenge.challengeRule;
+    } else {
+      this.adaptiveMfaRule = '';
+      this.challenge.challengeRule = '';
+    }
+    this.settingsChange.emit({
+      active: true,
```

Comment:
**issue**: this boolean shouldn't be force to true but should reflect the user decision when he enabled or disabled the toggle.